Basic-Car-Maintenance/Shared/Odometer/Views/OdometerView.swift

```diff
@@ -86,11 +121,37 @@ struct OdometerView: View {
                     }
                 }
             }
-
         }
         .analyticsView("\(Self.self)")
     }
     
+    // Helper function to filter readings based on the selected time range
+    private func filteredReadings(for vehicle: Vehicle) -> [OdometerReading] {
+        let vehicleReadings = viewModel.readings.filter { $0.vehicleID == vehicle.id }
+        
+        switch selectedTimeRange {
+        case .years:
+            return vehicleReadings
+        case .last30Days:
+            guard let lastReadingDate = vehicleReadings.map({ $0.date }).max() else {
+                return []
+            }
+            let thirtyDaysAgo = Calendar.current.date(byAdding: .day, value: -30, to: Date()) ?? Date()
+            
+            // If the last reading is older than 30 days, include only the last reading
+            if lastReadingDate < thirtyDaysAgo {
+                if let lastReading = vehicleReadings.max(by: { $0.date < $1.date }) {
+                    return [lastReading]
+                } else {
+                    return []
+                }
+            } else {
+                // Include readings from the last 30 days
```

Comment:
remove these extra comments because they aren't needed, the code is saying the same thing